src/mam4xx/ndrop.hpp

```diff
@@ -93,5 +108,48 @@ void explmix(
   }
 }
 
+// calculates maximum supersaturation for multiple
+// competing aerosol modes.
+// Abdul-Razzak and Ghan, A parameterization of aerosol activation.
+// 2. Multiple aerosol types. J. Geophys. Res., 105, 6837-6844.
+KOKKOS_INLINE_FUNCTION
+void maxsat(Real zeta, // [dimensionless]
+            Real eta[AeroConfig::num_modes()], // [dimensionless] 
+            Real nmode, //number of modes
+            Real smc[AeroConfig::num_modes()], // critical supersaturation for number mode radius [fraction]
+            Real smax //maximum supersaturation [fraction]
+            ) {
+
+    Real sum = 0;
+    Real g1, g2;
+    bool weak_forcing = true; // whether forcing is sufficiently weak or not
+
+    for(int m = 0; m < nmode; m++) {
+        if(zeta > 1e5 * eta[m] || smc[m] * smc[m] > 1e5 * eta[m]) {
+            // weak forcing. essentially none activated
+            smax = 1e-20;
```

Comment:
not 100% positive that this e syntax works in c++

Reply:
Do you mean the lack of parenthesis in the `if` expression? I think logical operators have lower precedence than ordering operators. Of course, adding parens usually improves clarity.